packages/server/src/fhir/repo.ts

```diff
@@ -1380,31 +1376,32 @@ export class Repository extends BaseRepository implements FhirRepository {
    * If it is a public resource type, then returns the public project ID.
    * If it is a protected resource type, then returns the Medplum project ID.
    * Otherwise, by default, return the current context project ID.
-   * @param resource - The FHIR resource.
+   * @param existing - Existing resource if one exists.
+   * @param updated - The FHIR resource.
    * @returns The project ID.
    */
-  private getProjectId(resource: Resource): string | undefined {
-    if (resource.resourceType === 'Project') {
-      return resource.id;
+  private getProjectId(existing: Resource | undefined, updated: Resource): string | undefined {
+    if (updated.resourceType === 'Project') {
+      return updated.id;
     }
 
-    if (resource.resourceType === 'ProjectMembership') {
-      return resolveId(resource.project);
+    if (updated.resourceType === 'ProjectMembership') {
+      return resolveId(updated.project);
     }
 
-    if (protectedResourceTypes.includes(resource.resourceType)) {
+    if (protectedResourceTypes.includes(updated.resourceType)) {
       return undefined;
     }
 
-    const submittedProjectId = resource.meta?.project;
+    const submittedProjectId = updated.meta?.project;
     if (submittedProjectId && this.canWriteMeta()) {
       // If the resource has an project (whether provided or from existing),
       // and the current context is allowed to write meta,
       // then use the provided value.
       return submittedProjectId;
     }
 
-    return this.context.project;
+    return existing?.meta?.project ?? this.context.project;
```

Comment:
@codyebberson I believe this is the desired order of preference. There's a slight logic change that results from this though: previously, an unchanged projectID would have been pulled into `submittedProjectId` above given the way meta from existing and updated were being merged. That arguably seems like it was semantically incorrect, but I think it points to a bigger issue that `canWriteMeta` is too broad since we want to allow non-superadmins to edit `meta.profile`.

Should `canWriteMeta` be renamed to `canWriteProtectedMeta` or something along those lines?

Reply:
> I believe this is the desired order of preference.

Hmm 🤔 I'm trying to think what would be the setup where the order would matter...

When would `existing?.meta?.project` be `undefined`, but `this.context.project` has a value?

`existing?.meta?.project` is only `undefined` for resources outside of a `Project`, which generally means super admin resources, which means only super admins should ever be able to read/write them.

`this.context.project` is always set for all users other than super admins and `systemRepo`.

I think... I think order doesn't matter here, and they should never be in conflict?  So I think the way you have it is ok 👍 

> Should canWriteMeta be renamed to canWriteProtectedMeta or something along those lines?

Yes, I think that better represents the intent of the method.